### PR TITLE
Natalia Malinowska - Etap 6, React Advanced Tests

### PR DESCRIPTION
**Pull request by natmali404**
should be done much earlier, but thanks to my genius i have invented a **GREAT** test case called "should log in and show plans list".
so it took me hours to try and basically create a E2E test in one test case for this assignment :") my bad. i misunderstood the assignment. (and wasted a lot of time.)

also yes. everything is in one file. sue me for this.

### CODE REVIEW

File: 2a. React/3. Advanced tests/src/pages/login.test.tsx [hunk 1]
```diff
@@ -1,25 +1,126 @@
...
+
+  it("should validate incorrect or foreign email", async () => {
+    mockUseAuth(false);
     const screen = render(<LoginPage />, {
       wrapper: Providers,
     });
     const user = userEvent.setup();
 
-    const emailInput = screen.getByLabelText("Adres e-mail");
+    const emailInput = screen.getByLabelText(/Adres e-mail/i);
```
Comment (czaja307) [inline 2a. React/3. Advanced tests/src/pages/login.test.tsx:32]: praise: the student has become the master 🙏  correcting Bartek's code is an achievement

File: 2a. React/3. Advanced tests/src/pages/login.test.tsx [hunk 2]
```diff
@@ -1,25 +1,126 @@
...
 
     const submitButton = screen.getByRole("button");
 
-    await user.type(emailInput, "invalid-email");
+    //not sure if i can test both these cases in one test or i should split them
+    //1
```
Comment (czaja307) [inline 2a. React/3. Advanced tests/src/pages/login.test.tsx:37]: Personally I'd opt for splitting it into 2 test cases. Why? If you leave it as is, when something breaks - you have to analyse logs to check which edge case failed. Otherwise you can just look at TC name to determine which one is failing.

File: 2a. React/3. Advanced tests/src/tests/mocks/handlers.ts [hunk 3]
```diff
@@ -1,3 +1,58 @@
-import type { RequestHandler } from "msw";
+import { http, HttpResponse } from "msw";
+import { BASE_URL } from "@/api/base-url";
 
-export const handlers = [] satisfies RequestHandler[];
+const mockOtp = "420420";
```
Comment (czaja307) [inline 2a. React/3. Advanced tests/src/tests/mocks/handlers.ts:4]: blaze it 🍃

File: 2a. React/3. Advanced tests/src/pages/login.test.tsx [hunk 4]
```diff
@@ -1,25 +1,126 @@
...
+  //   const screen = render(
+  //     <Providers>
+  //       <RouterProvider router={router} />
+  //     </Providers>
+  //   );
+
+  //   await waitFor(() => {
+  //     expect(screen.getByLabelText(/adres e-mail/i)).toBeInTheDocument();
+  //   });
+  // });
```
Comment (czaja307) [inline 2a. React/3. Advanced tests/src/pages/login.test.tsx:125]: i'd try to start with different approach to mock router but it's only my first show

let me know if you want deeper analysis

File: 2a. React/3. Advanced tests/src/tests/mocks/handlers.ts [hunk 5]
```diff
@@ -1,3 +1,58 @@
...
+
+//without it ts was acting up, idk if this is the correct way...
+type OtpRequest = {
+  email: string;
+};
```
Comment (czaja307) [inline 2a. React/3. Advanced tests/src/tests/mocks/handlers.ts:9]: seems A-okay to me

**Review by czaja307 — APPROVED** [review]
:fire: let's gooo!!
